feat: add bigframes.options.compute.maximum_bytes_billed option that sets maximum bytes billed on query jobs

orrbradford opened the pull request:
-implement context manager for global options
-maximum_bytes_billed only applies to query jobs. This limitation will be set per query. Operations that trigger multiple jobs may result in total usage beyond this setting

orrbradford committed:
feat: add `bigframes.options.compute.maximum_bytes_billed` option that sets maximum bytes billed on query jobs

-implement context manager for global options
-maximum_bytes_billed only applies to query jobs. This limitation will be set per query. Operations that trigger multiple jobs may result in total usage beyond this setting

orrbradford committed:
feat: add `bigframes.options.compute.maximum_bytes_billed` option that sets maximum bytes billed on query jobs

-implement context manager for global options
-maximum_bytes_billed only applies to query jobs. This limitation will be set per query. Operations that trigger multiple jobs may result in total usage beyond this setting

tswast commented on third_party/bigframes_vendored/pandas/_config/config.py:
```diff
@@ -0,0 +1,44 @@
+import contextlib
```
Needs license header.

orrbradford replied:
Done. Add a reference to the pandas file similar to other files in the parent directory. Let me know if this works

orrbradford committed:
add liscence header

orrbradford committed:
Merge branch 'main' into orrbradford-job-dev

orrbradford committed:
update test to pass coverage check

orrbradford committed:
Merge branch 'orrbradford-job-dev' of github.com:googleapis/python-bigquery-dataframes into orrbradford-job-dev

orrbradford commented:
kokoro rerun

orrbradford committed:
Merge branch 'main' into orrbradford-job-dev

tswast committed:
Merge branch 'main' into orrbradford-job-dev

ashleyxuu committed:
docs: add artithmetic df sample code (#153)

* docs: add artithmetic df sample code

* fix: address comments

tswast commented on bigframes/pandas/__init__.py:
```diff
@@ -175,6 +175,13 @@ def merge(
     )
 
 
+options = config.options
```
This already exists at the bottom of this file. See: https://github.com/googleapis/python-bigquery-dataframes/blob/ac44ccd3936cdb28755d2bbe16377d489f08d5e5/bigframes/pandas/__init__.py#L462

orrbradford replied:
Done

tswast commented on bigframes/pandas/__init__.py:
```diff
@@ -175,6 +175,13 @@ def merge(
+options = config.options
+"""Global :class:`~bigframes._config.Options` to configure BigQuery DataFrames."""
+
+option_context = config.option_context
```
Need to add this to `__all__`: https://github.com/googleapis/python-bigquery-dataframes/blob/ac44ccd3936cdb28755d2bbe16377d489f08d5e5/bigframes/pandas/__init__.py#L472

Would also be good to defined below near the `options` definition.

orrbradford replied:
Done

shobsi committed:
feat: Implement operator `@` for `DataFrame.dot` (#139)

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://togithub.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes b/297502513 🦕

ashleyxuu committed:
test: add code snippets for loading data from BigQuery Job (#154)

* test: add code snippets for loading data from BigQuery Job

* fix: address the comments

* fix: fix the broken test

* use BigQuery Client library to get the job_id

* feat: Implement operator `@` for `DataFrame.dot` (#139)

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://togithub.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes b/297502513 🦕

* fix: fix the comments

---------

Co-authored-by: Shobhit Singh <shobs@google.com>

orrbradford committed:
add options_context to __all__

orrbradford committed:
Merge branch 'main' of github.com:googleapis/python-bigquery-dataframes into orrbradford-job-dev

tswast commented on bigframes/_config/compute_options.py:
```diff
@@ -0,0 +1,35 @@
+
+
+@dataclasses.dataclass
+class ComputeOptions:
```
Could you add an "autoclass" for this file.

Need to update the following files for this to actually get documented:

* https://github.com/googleapis/python-bigquery-dataframes/blob/main/docs/reference/bigframes/options.rst
* https://github.com/googleapis/python-bigquery-dataframes/blob/main/docs/templates/toc.yml

orrbradford replied:
Done

tswast reviewed (commented):
Looking good, just one thing I forgot about so that we get this in the docs.

orrbradford committed:
add autoclass for compute options

orrbradford committed:
update bad autoclass config
